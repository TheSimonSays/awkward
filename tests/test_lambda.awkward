let double = lambda x: x * 2;
print(double(5));

let add = lambda x, y: x + y;
print(add(3, 7));

let greet = lambda name: "Hello, " + name;
print(greet("World"));

let calc = lambda x, y: (x + y) * 2;
print(calc(5, 3));

let nums = [1, 2, 3, 4, 5];
let square = lambda x: x * x;
print(square(4));

let multiplier = 10;
let multiply = lambda x: x * multiplier;
print(multiply(5));

fn apply(func, value) {
    return func(value);
}
let triple = lambda x: x * 3;
print(apply(triple, 7));

let addOne = lambda x: x + 1;
print(addOne(addOne(addOne(0))));
