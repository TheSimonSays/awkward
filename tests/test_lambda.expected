10
10
Hello, World
16
16
50
21
3
